### PR TITLE
Add MELT Dockerfile + build script

### PR DESCRIPTION
**Pull request by pdiakumis**

### CODE REVIEW

**Comment by pdiakumis** [comment]
Initially set out to have our own Docker images on Google Artifact Registry in the australia-southeast1 region, but seems that Cromwell doesn't support this registry. Have now reverted to Google Container Registry (in the US region). Not sure what the drawbacks (in a GCP world) would be if we simply used biocontainers from https://quay.io/ (automatically released via the bioconda build process), or if it's simply cheaper to host our own.

**Comment by pdiakumis** [comment]
@lgruen I've been thinking about this a bit more. I don't think we need to keep the Dockerfiles around for the main tools that already have images built and available on Quay e.g.
- Delly: https://quay.io/repository/biocontainers/delly?tab=tags
- Manta: https://quay.io/repository/biocontainers/manta?tab=tags
- wham: https://quay.io/repository/biocontainers/wham?tab=tags

We can simply copy the latest tags over to `us.gcr.io` with `docker pull` and `docker push`, and we don't even need to worry about builds/GitHub Actions. What do you think?

I'll keep the Melt Dockerfile + build script since it's not available publicly and will put the tarball in a private bucket for us.

**Comment by lgruen** [comment]
> @lgruen I've been thinking about this a bit more. I don't think we need to keep the Dockerfiles around for the main tools that already have images built and available on Quay e.g.
> 
> * Delly: https://quay.io/repository/biocontainers/delly?tab=tags
> * Manta: https://quay.io/repository/biocontainers/manta?tab=tags
> * wham: https://quay.io/repository/biocontainers/wham?tab=tags
> 
> We can simply copy the latest tags over to `us.gcr.io` with `docker pull` and `docker push`, and we don't even need to worry about builds/GitHub Actions. What do you think?
> 
> I'll keep the Melt Dockerfile + build script since it's not available publicly and will put the tarball in a private bucket for us.

Sounds like a good plan! Copying the images would still be good, so we're isolated from any issues on Quay. It might also be a little faster to download images if we stay within GCP.

File: dockerfiles/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Pre-download MELT tarball (license restricted).
+# Keeping .gitignore at this level to avoid
+# cloud build ignoring this file when uploading.
+melt/MELTv2.2.2.tar.gz
```
[the remark this answers is not in the document]
Comment (pdiakumis) [reply dockerfiles/.gitignore:4]: @lgruen done - I've created a new bucket `cpg-sv-workflows` in my dev account.

**Comment by pdiakumis** [comment]
@lgruen I've cleaned out everything apart from the MELT docker build files. `docker push` instead of `gcloud builds submit` works fine - the latter requires the relevant cloudbuild service account to be given `storage.buckets.get` permissions.

File: dockerfiles/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (lgruen) [reply dockerfiles/.gitignore:4]: Would a bucket in `cpg-common` (potentially even `cpg-reference`) be more appropriate? We should avoid dependencies from our production workflows to dev projects.

**Review by lgruen — CHANGES_REQUESTED** [review]

File: dockerfiles/.gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdiakumis) [reply dockerfiles/.gitignore:4]: Sure, agreed! I don't have permissions to create that though, so let me know and we can add service account permissions.

**Pull request closed without merging**